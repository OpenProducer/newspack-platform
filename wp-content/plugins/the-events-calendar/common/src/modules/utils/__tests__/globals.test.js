/**
 * Internal dependencies
 */
import {
	google,
	mapsAPI,
	settings,
	list,
	get,
	config,
	rest,
	restNonce,
} from '@moderntribe/common/utils/globals';

describe( 'Tests for globals.js', () => {

	beforeAll( () => {
		window.tribe_editor_config = {
			common: {
				countries: {},
				usStates: {},
				settings: {},
				rest: {
					namespaces: {
						core: 'wp/v2',
					},
					nonce: {
						wp_rest: 'cedcd6967b',
						add_ticket_nonce: '0878f40fb2',
					},
					url: 'http://gutenberg.local/wp-json/',
				},
			},
			tec: {
				googleMap: {},
			},
			tickets: {
				providers: [ {
					class: 'Tribe__Tickets_Plus__Commerce__WooCommerce__Main',
					currency: '$',
					currency_position: 'prefix',
					name: 'WooCommerce',
				} ],
				default_provider: 'Tribe__Tickets_Plus__Commerce__WooCommerce__Main',
				default_currency: '$',
			},
		};
	} );

	test( 'Should match the default value for the globals values', () => {
		expect( get( 'random' ) ).toBe( undefined );
		expect( get( 'google' ) ).toBe( undefined );
		expect( google() ).toBe( undefined );
		expect( get( 'tribe_editor_config' ) ).toMatchSnapshot();
		expect( settings() ).toEqual( {} );
		expect( mapsAPI() ).toEqual( {} );
		expect( list() ).toEqual( {
			countries: {},
			us_states: {},
		} );
		expect( config() ).toMatchSnapshot();
	} );

	test( 'get default value', () => {
		expect( get( 'UNKNOWN', 10 ) ).toBe( 10 );
		expect( get( 'tribe_js_config', [] ) ).toMatchSnapshot();
	} );

	test( 'rest value', () => {
		expect( rest() ).toMatchSnapshot();
		expect( restNonce() ).toMatchSnapshot();
	} );

	afterAll( () => {
		delete window.tribe_editor_config;
	} );
} );

describe( 'Test default values on globals', () => {
	test( 'rest default values', () => {
		expect( rest() ).toEqual( {} );
		expect( restNonce() ).toEqual( {} );
	} );
} );
