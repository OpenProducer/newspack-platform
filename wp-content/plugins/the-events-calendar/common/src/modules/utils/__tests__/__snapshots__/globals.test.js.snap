// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for globals.js Should match the default value for the globals values 1`] = `
Object {
  "common": Object {
    "countries": Object {},
    "rest": Object {
      "namespaces": Object {
        "core": "wp/v2",
      },
      "nonce": Object {
        "add_ticket_nonce": "0878f40fb2",
        "wp_rest": "cedcd6967b",
      },
      "url": "http://gutenberg.local/wp-json/",
    },
    "settings": Object {},
    "usStates": Object {},
  },
  "tec": Object {
    "googleMap": Object {},
  },
  "tickets": Object {
    "default_currency": "$",
    "default_provider": "Tribe__Tickets_Plus__Commerce__WooCommerce__Main",
    "providers": Array [
      Object {
        "class": "Tribe__Tickets_Plus__Commerce__WooCommerce__Main",
        "currency": "$",
        "currency_position": "prefix",
        "name": "WooCommerce",
      },
    ],
  },
}
`;

exports[`Tests for globals.js Should match the default value for the globals values 2`] = `
Object {
  "common": Object {
    "countries": Object {},
    "rest": Object {
      "namespaces": Object {
        "core": "wp/v2",
      },
      "nonce": Object {
        "add_ticket_nonce": "0878f40fb2",
        "wp_rest": "cedcd6967b",
      },
      "url": "http://gutenberg.local/wp-json/",
    },
    "settings": Object {},
    "usStates": Object {},
  },
  "tec": Object {
    "googleMap": Object {},
  },
  "tickets": Object {
    "default_currency": "$",
    "default_provider": "Tribe__Tickets_Plus__Commerce__WooCommerce__Main",
    "providers": Array [
      Object {
        "class": "Tribe__Tickets_Plus__Commerce__WooCommerce__Main",
        "currency": "$",
        "currency_position": "prefix",
        "name": "WooCommerce",
      },
    ],
  },
}
`;

exports[`Tests for globals.js get default value 1`] = `Array []`;

exports[`Tests for globals.js rest value 1`] = `
Object {
  "namespaces": Object {
    "core": "wp/v2",
  },
  "nonce": Object {
    "add_ticket_nonce": "0878f40fb2",
    "wp_rest": "cedcd6967b",
  },
  "url": "http://gutenberg.local/wp-json/",
}
`;

exports[`Tests for globals.js rest value 2`] = `
Object {
  "add_ticket_nonce": "0878f40fb2",
  "wp_rest": "cedcd6967b",
}
`;
